build-system/tasks/prettify.js

```diff
@@ -119,8 +119,13 @@ function printFixMessages() {
  */
 async function runPrettify(filesToCheck) {
   logLocalDev(green('Starting checks...'));
+  const trackedFilesToCheck = getStdout(
+    ['git ls-files', ...filesToCheck].join(' ')
+  )
+    .trim()
+    .split('\n');
```

Comment:
I think this is unrelated to the `prettify` logic, and should live in `getFilesToCheck()` in `utils.js`.

And I believe that function already provides a mechanism to ignore different kinds of files like untracked files, files in `.gitignore`, etc. by passing in an options object. E.g. [`{gitignore: true}`](https://github.com/sindresorhus/globby#gitignore). There are other options that can do this, I'm assuming.

Reply:
Should we then include `--local_changes` here?

https://github.com/ampproject/amphtml/blob/e32ced7db8d20ddbca757807360e822e8af608a7/build-system/pr-check/checks.js#L62-L64

```js
 if (buildTargetsInclude(Targets.PRETTIFY)) { 
   timedExecOrDie('gulp prettify --local_changes'); 
 } 
 ```

Reply:
We tried this in the past, but there were instances of conflicting PRs resulting in breaking changes to slip through to `master`. The task doesn't take long to run, so we should continue to check all files.

Reply:
What would be the best solution? I find it dangerous if files generated by CI can be included in the diff here.

Reply:
Agree with your concern, and glad you're trying to solve this 😄

There are a couple solutions:
- Make the glob pattern for `prettify` less generic (line 167 in particular)
  https://github.com/ampproject/amphtml/blob/e340dad5f10f99c7f6d9aca7dcf30ced387009cf/build-system/test-configs/config.js#L151-L173
- Pass in an `options` object to `getFilesToCheck()` to make it ignore untracked files and files in `.gitignore` (`globby` has a few [options](https://github.com/sindresorhus/globby#options) of its own, and also supports all [options](https://github.com/mrmlnc/fast-glob#options-3) from `fast-glob`)

Reply:
> Pass in an options object to getFilesToCheck() to make it ignore untracked files and files in .gitignore (globby has a few options of its own, and also supports all options from fast-glob)

I went for this option, please take a look.